test/openAPI/features/support/helpers/helpers.js

```diff
@@ -1,7 +1,181 @@
+              ID: {
+                type: 'object',
+                properties: {
+                  type: 'string',
+                  triggers: {
+                    type: 'array',
+                    items: {
+                      type: 'object',
+                      properties: {
+                        conditions: {
+                          type: 'array',
+                          properties: {
+                            logic: {
+                              type: 'string',
+                            },
+                            value: {
+                              type: 'string',
+                            },
+                            gate: { type: 'string' },
+                          },
+                          additionalProperties: false,
+                        },
+                        actions: {
+                          type: 'array',
+                          items: {
+                            type: 'object',
+                            properties: {
+                              type: {
+                                type: 'string',
+                              },
+                              value: { type: 'string' },
+                              field_id: { type: 'integer' },
+                            },
+                            additionalProperties: false,
+                          },
+                        },
+                      },
+                      additionalProperties: false,
+                    },
+                  },
+                  primaryKey: { type: 'boolean' },
+                  readOnly: { type: 'boolean' },
+                  description: { type: 'string' },
+                  example: { type: 'string' },
+                  id: { type: 'integer' },
+                },
+                additionalProperties: false,
+              },
+              Child: {
+                type: 'object',
+                properties: {
+                  type: {
+                    type: 'string',
+                    properties: {
+                      type: 'object',
+                      properties: {
+                        ID: {
+                          type: 'object',
+                          properties: {
+                            type: { type: 'string' },
+                            description: {
+                              type: 'string',
+                            },
+                            example: {
+                              type: 'string',
+                            },
+                            id: {
+                              type: 'integer',
+                            },
+                          },
+                        },
+                      },
+                    },
+                  },
+                },
+              },
+            },
+            additionalProperties: false,
+          },
+          incrementIndex: {
+            type: 'integer',
+          },
+          required: {
```

Comment:
If it's supposed to be an array of strings with required keys of fields that must be present I think we should create another ```And``` example in the .feature file to test it. I'm not totally sure how it should work, so it'd be best to ask bb-lead

Reply:
@ingmarvali what does ```required``` array do in this case.  Lets say it contains a value: for example 'ID'. Does that mean that the schema MUST contain the property ID? Or maybe schema and its Child must contain the field?
let me know please if my thoughts on it make sense.  Here is the schema: 
![image](https://user-images.githubusercontent.com/119664031/227030655-3175d9e4-022e-4658-88a5-262cd294059a.png)

Reply:
the required array contains a list of field names that are required in this database, meaning the data is required in the input, not in the schema.

Reply:
Properties are dynamic if this is your question. here is an example from second database/registry.
{
	"type": "object",
	"properties": {
		"ID": {
			"type": "string",
			"triggers": [{
				"conditions": [{
					"logic": "==",
					"value": "",
					"gate": "&&"
				}],
				"actions": [{
					"type": "set-value",
					"value": "{generateUUID}",
					"field_id": 1
				}, {
					"type": "upper-case",
					"field_id": 1
				}]
			}],
			"primaryKey": true,
			"readOnly": true,
			"description": "Registry ID",
			"example": "E71691A2-6299-44FF-A1E9-29AA09FA78C5",
			"$id": 1,
			"isFullText": true,
			"tags": "",
			"flags": "unique,mandatory"
		},
		"Citizen ID": {
			"type": "string",
			"triggers": [{
				"conditions": [{
					"logic": "*",
					"value": "",
					"gate": "&&"
				}],
				"actions": [{
					"type": "set-value",
					"value": "{indexNoByCode}",
					"field_id": 2
				}]
			}],
			"sortable": true,
			"readOnly": true,
			"description": "Unique personal identifier. Source: national citizen registry or population registry. ",
			"example": "1",
			"$id": 2,
			"isFullText": true,
			"tags": "",
			"flags": "unique"
		},
		"First names": {
			"type": "string",
			"description": "First names",
			"example": "Usha",
			"$id": 3,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Last names": {
			"type": "string",
			"triggers": [{
				"conditions": [{
					"logic": "*",
					"value": "",
					"gate": "&&"
				}],
				"actions": [{
					"type": "set-value",
					"value": "{field-3} {field-4}",
					"field_id": 8
				}]
			}],
			"description": "Last names",
			"example": "Bajaj",
			"$id": 4,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Full name": {
			"type": "string",
			"description": "First names and last names all in one",
			"example": "Usha Bajaj",
			"$id": 8,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Birthdate": {
			"type": "string",
			"format": "date",
			"description": "Date of birth",
			"$id": 5,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Death": {
			"type": "string",
			"format": "date",
			"$id": 23,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Nationality": {
			"type": "catalog",
			"properties": {
				"key": {
					"type": "string",
					"$id": 10,
					"tags": ""
				},
				"value": {
					"type": "string",
					"$id": 11,
					"tags": ""
				}
			},
			"required": ["key", "value"],
			"description": "Nationality",
			"$id": 9,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Nationality value": {
			"type": "string",
			"$id": 24,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Nationality key": {
			"type": "string",
			"$id": 21,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Gender": {
			"type": "string",
			"description": "Gender (Female, Male)",
			"example": "Female",
			"$id": 12,
			"isFullText": true,
			"tags": "",
			"flags": ""
		},
		"Mother": {
			"type": "object",
			"properties": {
				"Citizen ID": {
					"type": "string",
					"$id": 16,
					"isFullText": true,
					"tags": "",
					"flags": ""
				},
				"First names": {
					"type": "string",
					"$id": 14,
					"isFullText": true,
					"tags": "",
					"flags": ""
				},
				"Last names": {
					"type": "string",
					"$id": 15,
					"isFullText": true,
					"tags": "",
					"flags": ""
				}
			},
			"$id": 13,
			"tags": "",
			"flags": ""
		},
		"Father": {
			"type": "object",
			"properties": {
				"Citizen ID": {
					"type": "string",
					"$id": 20,
					"isFullText": true,
					"tags": "",
					"flags": ""
				},
				"First names": {
					"type": "string",
					"$id": 18,
					"isFullText": true,
					"tags": "",
					"flags": ""
				},
				"Last names": {
					"type": "string",
					"$id": 19,
					"isFullText": true,
					"tags": "",
					"flags": ""
				}
			},
			"$id": 17,
			"tags": "",
			"flags": ""
		}
	},
	"$incrementIndex": 24,
	"required": ["ID"],
	"tags": "",
	"flags": ""
}

Reply:
this test endpoint should be open for examples: https://gdb.er4.ext.egovstack.net/swaggerx/